daemon/cmd/datapath.go

```diff
@@ -201,14 +201,9 @@ func (d *Daemon) syncEndpointsAndHostIPs() error {
 	specialIdentities := []identity.IPIdentityPair{}
 
 	if option.Config.EnableIPv4 {
-		addrs, err := d.datapath.LocalNodeAddressing().IPv4().LocalAddresses()
-		if err != nil {
-			log.WithError(err).Warning("Unable to list local IPv4 addresses")
-		}
-
-		for _, ip := range addrs {
+		err := d.datapath.LocalNodeAddressing().IPv4().MapLocalAddresses(func(ip net.IP) error {
 			if option.Config.IsExcludedLocalAddress(ip) {
-				continue
+				return nil
 			}
 
 			if len(ip) > 0 {
```

Comment:
This doesn't quite solve it for the ~64MB memory spike I was referring to in my previous review. The `syncEndpointsAndHostIPs` was designed to handle a couple dozen of IPs, not these many.

My initial suggestion was to modify the following code (from line 264):

```golang
	for _, ipIDPair := range specialIdentities {
		hostKey := node.GetIPsecKeyIdentity()
		isHost := ipIDPair.ID == identity.ReservedIdentityHost
		if isHost {
			added, err := lxcmap.SyncHostEntry(ipIDPair.IP)
			if err != nil {
				return fmt.Errorf("Unable to add host entry to endpoint map: %s", err)
			}
			if added {
				log.WithField(logfields.IPAddr, ipIDPair.IP).Debugf("Added local ip to endpoint map")
			}
		}
```

All the logic between that for loop could be moved into a dedicated function, `Foo`, and then we could perform the logic to iterate over the CIDR range after this for loop.
```golang
	for _, ipIDPair := range specialIdentities {
		Foo(ipIDPair)
	}
	mapSubnetAddresses(nodePortAddresses, Foo)
```